### PR TITLE
Fix some stackdriver bugs

### PR DESCRIPTION
**Pull request by icco**
## Proposed Changes

* Continuing on #949, this fixes the missing ochttp metrics.
* Fixes location, which comes back in a bad format from the metadata server.
* Increases the metric reporting time (doesn't fix the errors, but makes them less frequent)

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: pkg/observability/resource.go [hunk 1]
```diff
@@ -72,6 +73,12 @@ func NewStackdriverMonitoredResource(ctx context.Context, c *StackdriverConfig)
 	}
 	labels["location"] = region
 
+	// Metadata often returns the following: "projects/111111111111/regions/us-central1"
+	pieces := strings.Split(region, "/")
```
Comment (sethvargo) [inline pkg/observability/resource.go:77]: This feels fragile and could have subtle bugs if that format changes

Comment (icco) [reply pkg/observability/resource.go:77]: Yeah. I'm not sure what to do otherwise. I feel like a regex would be more fragile?

Comment (sethvargo) [reply pkg/observability/resource.go:77]: SplitN and checking against the other lengths? Or maybe returning/logging an error on mismatch?

Comment (icco) [reply pkg/observability/resource.go:77]: good idea, added error message.

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**